registry/src/types.rs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use proptest::prelude::*;
```

Comment:
I dislike that proptest is mixed into the source code here - any other way around?

Reply:
Currently, what I am trying to do is derive the `Arbitrary` trait on things like `PackageVersion`. This allows me to write tests with proptest like so:

```rust
#[proptest]
async fn some_test(arbitrary_version: PackageVersion) {
    // ...
}
```

This does mean that we need to have some amount of proptest-related code in here (to derive that). Unfortunately, we cannot put it elsewhere because of the orphan rule (we cannot derive `Arbitrary` in the `tests/` for example). 

There is another way, which is using manual proptest strategies that we could employ which mean that we do not need to derive `Arbitrary`. It would look like this:

```
prop_compose! {
    fn package_version() -> PackageVersion {
        // create package version
    }
}

#[proptest]
async fn some_test(arbitrary_version in package_version()) {
    // ...
}
```

Now we could put that `prop_compose` into the tests folder.

Reply:
i would put the prop compone in the tests folder!
for the derive – can we feature flag it with the `test` feature, so its clear that this is for testing purposes only?

Reply:
We have moved all of the tests into `tests/` folder, which means that they are integration tests. Integration tests are not built with the `test` feature (they compile the crate exactly as it would be compiled by a user). So, unfortunately we do not have any flag we can check :/

I think the only option is to simply not derive `Arbitrary`, but implement the strategy manually.

Reply:
Fixed this by not deriving Arbitrary.